com.unity.shadergraph/Editor/Data/Interfaces/IInspectable.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Reflection;
+using Drawing.Inspector;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ShaderGraph.Drawing
+{
+    [AttributeUsage(AttributeTargets.Property)]
+    public class Inspectable : Attribute
+    {
+        // String value to use in the Property name TextLabel
+        public string labelName { get; private set; }
+
+        // The default value of this property
+        public object defaultValue { get; private set; }
+
+        // String value to supply if you wish to use a custom style when drawing this property
+        public string customStyleName { get; private set; }
+
+        public Inspectable(string labelName, object defaultValue, string customStyleName = "")
+        {
+            this.labelName = labelName;
+            this.defaultValue = defaultValue;
+            this.customStyleName = customStyleName;
+        }
+    }
+
+    interface IInspectable
+    {
+        string displayName { get; }

```

Comment:
Since most of the classes that implement this interface are already Views, putting "DisplayName" in them isn't clear what it's for. I think calling this "inspectorTitle" or "settingsTitle" or something would be clearer.